### PR TITLE
Detect database name from property file

### PR DESCRIPTION
**Pull request by rujche**
Detect database name from property file.

All types of database has been considered:
1. MySQL, PostgreSQL, MongoDB, Cosmos DB support detect database name from property file. 
2. Redis: connect to Redis don't need a database name.

### CODE REVIEW

File: cli/azd/internal/appdetect/spring_boot.go [hunk 1]
```diff
@@ -252,24 +252,50 @@ func detectStorageAccountAccordingToSpringCloudStreamBinderMavenDependencyAndPro
 
 func detectMetadata(azdProject *Project, springBootProject *SpringBootProject) {
 	detectPropertySpringApplicationName(azdProject, springBootProject)
+	detectPropertySpringCloudAzureCosmosDatabase(azdProject, springBootProject)
+	detectPropertySpringDataMongodbDatabase(azdProject, springBootProject)
+	detectPropertySpringDataMongodbUri(azdProject, springBootProject)
 	detectPropertySpringDatasourceUrl(azdProject, springBootProject)
```
Comment (rujche) [inline cli/azd/internal/appdetect/spring_boot.go:258]: Added 3 new methods.

File: cli/azd/internal/appdetect/spring_boot.go [hunk 2]
```diff
@@ -252,24 +252,50 @@ func detectStorageAccountAccordingToSpringCloudStreamBinderMavenDependencyAndPro
...
+
 	detectDependencySpringCloudAzureStarter(azdProject, springBootProject)
-	detectDependencySpringCloudAzureStarterJdbcPostgresql(azdProject, springBootProject)
 	detectDependencySpringCloudAzureStarterJdbcMysql(azdProject, springBootProject)
-	detectDependencySpringCloudEureka(azdProject, springBootProject)
+	detectDependencySpringCloudAzureStarterJdbcPostgresql(azdProject, springBootProject)
 	detectDependencySpringCloudConfig(azdProject, springBootProject)
+	detectDependencySpringCloudEureka(azdProject, springBootProject)
```
Comment (rujche) [inline cli/azd/internal/appdetect/spring_boot.go:264]: Just sort the methods by alphabet order.

File: cli/azd/internal/appdetect/spring_boot_property.go [hunk 3]
```diff
@@ -134,5 +135,8 @@ func getEnvironmentVariablePlaceholderHandledValue(rawValue string) string {
 		placeholder := match[0]
 		result = strings.Replace(result, placeholder, value, -1)
 	}
+	if result == "#{null}" {
+		result = ""
+	}
```
Comment (rujche) [inline cli/azd/internal/appdetect/spring_boot_property.go:140]: Originally, I add this to support this scenario:
> spring.data.mongodb.uri=${AZURE_COSMOS_CONNECTION_STRING:#{null}}

`#{null}` is SPEL.

This idea comes from here:
https://github.com/Azure-Samples/todo-java-mongo-aca/blob/d8fff510c9185c6ec4754a6671c2776a51b5668d/src/api/src/main/resources/application.properties#L10
> image

But when I debug it, I found that SPEL is not supported in `DataObjectPropertyBinder`.
> image

I'll remove this.

File: cli/azd/internal/repository/infra_confirm.go [hunk 4]
```diff
@@ -226,7 +226,7 @@ func getDatabaseNameFromProjectMetadata(detect *detectConfirm, database appdetec
 	result := ""
 	for _, service := range detect.Services {
 		name := service.Metadata.DatabaseNameInPropertySpringDatasourceUrl[database]
-		if name != result {
+		if name != "" {
```
Comment (rujche) [inline cli/azd/internal/repository/infra_confirm.go:229]: This is a bug before current PR: In multi-module project, one module configured database name, and another module, the project name is not configured, the detected database name will not be used before current PR.

**Comment by rujche** [comment]
Hi, @saragluna , @haoozhang . Please help to review this PR.

**Review by haoozhang — APPROVED** [review]

**Pull request merged**